pkg/block/hashing_reader.go

```diff
@@ -39,7 +39,7 @@ func (s *HashingReader) Read(p []byte) (int, error) {
 func NewHashingReader(body io.Reader, hashTypes ...int) *HashingReader {
 	s := new(HashingReader)
 	s.originalReader = body
-	for hashType := range hashTypes {
+	for _, hashType := range hashTypes {
```

Comment:
Nice catch - glad that the value of each hash type match the value of the of hash type, producing MD5 and/or SHA256 relative to the value passing to this function.

Reply:
done